### ISSUE TITLE
token.to_s throwing an error

### ISSUE THREAD

**Issue by pierreyves-lebrun** [codesthq/jwt-authorizer#12]
Hello,

sorry if am using the gem incorrectly but I am getting the following error upon calling to_s on my token:

```
#<ApiToken:0x00007fa5103335d0 @claims={"exp"=>1541978086, "iss"=>nil, "user_id"=>2}>
2018-11-11 20:14:46 - NoMethodError - undefined method `sign' for nil:NilClass:
	/Users/pierreyves.lebrun/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/jwt-2.1.0/lib/jwt/algos/rsa.rb:11:in `sign'
	/Users/pierreyves.lebrun/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/jwt-2.1.0/lib/jwt/signature.rb:35:in `sign'
	/Users/pierreyves.lebrun/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/jwt-2.1.0/lib/jwt/encode.rb:39:in `encoded_signature'
	/Users/pierreyves.lebrun/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/jwt-2.1.0/lib/jwt/encode.rb:47:in `encode_segments'
	/Users/pierreyves.lebrun/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/jwt-2.1.0/lib/jwt/encode.rb:20:in `initialize'
	/Users/pierreyves.lebrun/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/jwt-2.1.0/lib/jwt.rb:21:in `new'
	/Users/pierreyves.lebrun/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/jwt-2.1.0/lib/jwt.rb:21:in `encode'
	/Users/pierreyves.lebrun/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/jwt-authorizer-2.0.0/lib/jwt/token/builder.rb:11:in `to_jwt'
```

Below is the code I used to generate my token:

```
JWT::Token.configure do |config|
  config.expiry = 12 * 60 * 60
  config.algorithm = "RS256"
  config.rsa.authorized_keys = [OpenSSL::PKey::RSA.new(2048)]
end

class ApiToken < JWT::Token
  claim :user_id, required: true
end

token = ApiToken.new(user_id: 1)
token.to_s #Errors
```

**Comment by fanfilmu** [comment]
@pierreyves-lebrun The reason for this is that you didn't set up a private key for the token, which is required to sign the tokens.

The class as you configured it is only usable for verifying the signatures; if you want to create a signed token, you also need to configure `private_key` and `allowed_algorithms`:

```
the_key = OpenSSL::PKey::RSA.new(2048)

JWT::Token.configure do |config|
  config.expiry = 12 * 60 * 60
  config.algorithm = "RS256"
  config.rsa.authorized_keys = [the_key]
  
  config.rsa.private_key = the_key
  config.allowed_algorithms = ["RS256"]
end

class ApiToken < JWT::Token
  claim(:user_id, required: true) {}
end

token = ApiToken.new(user_id: 1)
jwt = token.to_s
ApiToken.verify(jwt)
```

Also had to add empty verifier for the claim... I'll work on error reporting here and allowing null verifier, but for the quick fix it should be OK :)

**Comment by fanfilmu** [comment]
#13 but I'll have to get ahold of somebody with write access which I lost :D

**Comment by radarek** [comment]
@fanfilmu Here you go :).

**Comment by pierreyves-lebrun** [comment]
Thanks guys, that all makes sense!

**Comment by fanfilmu** [comment]
@pierreyves-lebrun Released [2.0.1](https://rubygems.org/gems/jwt-authorizer/versions/2.0.1) with the mentioned fixes. Enjoy :)
